bindings/matrix-sdk-ffi/src/ruma.rs

```diff
@@ -985,3 +1018,591 @@ pub fn content_without_relation_from_message(
     let msg_type = message.msg_type.try_into()?;
     Ok(Arc::new(RoomMessageEventContentWithoutRelation::new(msg_type)))
 }
+
+/// Types of global account data events.
+#[derive(Clone, uniffi::Enum)]
+pub enum AccountDataEventType {
+    /// m.direct
+    Direct,
+    /// m.identity_server
+    IdentityServer,
+    /// m.ignored_user_list
+    IgnoredUserList,
+    /// m.push_rules
+    PushRules,
+    /// m.secret_storage.default_key
+    SecretStorageDefaultKey,
+    /// m.secret_storage.key.*
+    SecretStorageKey { key_id: String },
+}
```

Comment:
There's a lot of duplication of Ruma types here. I remember the Ruma folks being open towards adding `uniffi` attributes to Ruma types directly.

It's prbobably a big ask, but would you be open on attempting this?

The account data types might be sufficiently small that this might not be too big of a problem.

Reply:
> I remember the Ruma folks being open towards adding uniffi attributes to Ruma types directly.

Oh, I wasn't aware. That would actually be amazing. The mapping of Ruma to FFI types adds a lot of boilerplate in many places.

I can try to push things done. It might just take me a little. Just saying in case there's any urgency on https://github.com/matrix-org/matrix-rust-sdk/pull/5014 (or its successor).

Reply:
I think we can also merge this as is, and if/when `uniffi` attributes are added to the Ruma type we can just switch over.